design-proposals/vm_video_device_type_field.md

```diff
@@ -0,0 +1,77 @@
+# Overview
+We want to allow VM-owners to explicitly set the video device type when needed.
+
+## Motivation
+Currently, the default video device type for AMD64 in KubeVirt is **VGA** or **Bochs** in some cases.
+Using `vga` video type can limit the video functionality available to virtual machines.
+For example, VGA restricts display resolution settings for Windows VMs and may not support modern workloads effectively.
+However, it is necessary to maintain compatibility for older virtual machines that depend on VGA (e.g., RHEL 5/6).
+To address this, we propose introducing a new configuration field to allow users to set the video device type.
+
+## Goals
+The VM spec template resource should have the ability to set the video device type when needed.
+
+## Non Goals
+Change the defaulting logic
+
+## Definition of Users
+VM-owners who require specific configurations for VMIs in their environments.
+
+## User Stories
+- As a VM owner, I want to be able to set a better video device like `virtio` instead of the standard `vga`.
+
+## Repos
+- [KubeVirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+
+### VM-Level Configuration Field
+
+**Description:** Add a `Video` struct under `spec.template.spec.domain.devices` in the VM template schema.
+This struct for now, will include a single field, type, to specify the desired video device type.
+Currently, the virt-launcher arch-converter manages the video type.
+With this change, users can explicitly set the video type, and the arch-converter will respect it.
+
+**Pros:**
+- Granular Control: Enables precise configuration for each VM, avoiding cluster-wide changes that could affect unrelated workloads.
+- Default Logic: Ensures reasonable defaults based on architecture while allowing user overrides for legacy compatibility.
+- Scalability: Provides flexibility for administrators and VM owners in mixed environments with diverse requirements.
+
+**Cons:**
+- API Complexity: Requires schema changes to the VM template API, introducing additional fields that need maintenance and testing.
+- Administrative Effort: Per-VM configuration may lead to higher administrative overhead if many VMs require toggling.
+
+## API Examples
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          video:
+            type: virtio
```

Comment:
Libvirt supports multiple video devices [[1](https://libvirt.org/formatdomain.html#video-devices)]:
```xml
<devices>
  <video>
    <model type='vga'/>
  </video>
  <video>
    <model type='virtio'/>
  </video>
</devices>
```

They also have the "primary" field to indicate which is the primary volume.
Since that's the case, I think we want to turn the `video` field to a slice, and perhaps add the `primary` field. In libvirt, if the primary field is not supplied, than the first video device is set as the primary, perhaps we can follow this approach.